front50-web/src/test/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsControllerV2Tck.groovy

```diff
@@ -0,0 +1,356 @@
+  }
+
+  def "should update an application"() {
+    setup:
+    def owner = "Andy McEntee"
+    def sampleApp = new Application(name: "SAMPLEAPP", email: "web@netflix.com", owner: owner)
+    dao.create("SAMPLEAPP", new Application())
+
+    when:
+    def response = mockMvc
+      .perform(
+      put("/v2/applications/SAMPLEAPP")
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(new ObjectMapper().writeValueAsString(sampleApp))
+    )
+
+    then:
+    response.andExpect status().isOk()
+    toMap(dao.findByName("SAMPLEAPP")) == toMap(sampleApp)
+  }
+
+  def "should not update an application if no name is provided"() {
+    given:
+    def sampleApp = new Application(email: "web@netflix.com", owner: "Andy McEntee")
+
+    when:
+    def response = mockMvc
+      .perform(
+      put("/v2/applications")
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(new ObjectMapper().writeValueAsString(sampleApp))
+    )
+
+    then:
+    response.andExpect status().is4xxClientError()
+  }
+
+  def "should not update an application if provided name doesn't match the application name"() {
+    given:
+    def sampleApp = new Application(email: "web@netflix.com", name: "SAMPLEAPP")
+
+    when:
+    def response = mockMvc
+      .perform(
+      put("/v2/applications/SAMPLEAPPWRONG")
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(new ObjectMapper().writeValueAsString(sampleApp))
+    )
+
+    then:
+    response.andExpect status().is4xxClientError()
+  }
+
+  def "should merge properties with existing ones on update"() {
+    setup:
+    def app = "SAMPLEAPP"
+    def newEmail = "new@netflix.com"
+    def dynamicPropertyToUpdate = "new dynamic property"
+    def description = "old"
+    def unchangedDynamicProperty = "no change"
+    def brandNewDynamicProperty = "brand new"
+    def existingApp = new Application(
+      name: app,
+      email: "old@netflix.com",
+      description: description,
+      dynamicPropertyToUpdate: "old dynamic property",
+      unchangedDynamicProperty: unchangedDynamicProperty
+    )
+
+    dao.create(app, existingApp)
+
+    def updates = new Application(
+      name: app,
+      email: newEmail,
+      dynamicPropertyToUpdate: dynamicPropertyToUpdate,
+      brandNewDynamicProperty: brandNewDynamicProperty
+    )
+
+    when:
+    def response = mockMvc.perform(
+      put("/v2/applications/SAMPLEAPP")
+        .contentType(MediaType.APPLICATION_JSON)
+        .content(new ObjectMapper().writeValueAsString(updates))
+    )
+
+    then:
+    response.andExpect status().isOk()
+    toMap(dao.findByName(app)) == toMap(new Application(
+      name: app,
+      description: description,
+      email: newEmail,
+      dynamicPropertyToUpdate: dynamicPropertyToUpdate,
+      unchangedDynamicProperty: unchangedDynamicProperty,
+      brandNewDynamicProperty: brandNewDynamicProperty
+    ))
+  }
+
+  def "should find an application by name"() {
+    setup:
```

Comment:
Might find that `given` is more commonly used than `setup` but half dozen or the other really.